lib/Sema/CSGen.cpp

```diff
@@ -1991,12 +1991,9 @@ namespace {
         // tuples, nested arrays, and dictionary literals.
         //
         // Otherwise, create a new type variable.
-        auto boundExpr = locator.trySimplifyToExpr();
-
-        if (boundExpr) {
-          auto boundExprTy = CS.getType(boundExpr);
-          if (!boundExprTy->is<InOutType>())
-            return boundExprTy->getRValueType();
+        if (auto boundExpr = locator.trySimplifyToExpr()) {
+          if (!boundExpr->isSemanticallyInOutExpr())
+            return CS.getType(boundExpr)->getRValueType();
```

Comment:
@rudkx Was there a specific reason this wasn't using the type map before?

Reply:
Do you mean `CS.getType()`? That's just a couple more lines up in the diff.

Reply:
Ah, carry on then.